Zotero/Scenes/General/NoteEditorCoordinator.swift

```diff
@@ -112,4 +121,41 @@ extension NoteEditorCoordinator: NoteEditorCoordinatorDelegate {
         }
         navigationController?.present(controller, animated: true)
     }
+
+    func showPdf(withPreview preview: AnnotationPreview) {
+        showPdf(key: preview.parentKey, libraryId: preview.libraryId, preview: (preview.pageIndex, preview.rects))
+    }
+
+    func showPdf(withCitation citation: CitationMetadata) {
+        showPdf(key: citation.attachmentKey, libraryId: citation.libraryId, preview: (citation.locator, nil))
+    }
+
+    func showItemDetail(withCitation citation: CitationMetadata) {
+        guard let coordinator = (parentCoordinator as? DetailCoordinator) else { return }
+        coordinator.showItemDetail(for: .preview(key: citation.parentKey), libraryId: citation.libraryId, scrolledToKey: nil, animated: false)
+        navigationController?.dismiss(animated: true)
+    }
+
+    private func showPdf(key: String, libraryId: LibraryIdentifier, preview: (UInt, [CGRect]?)) {
```

Comment:
I'm guessing `preview` wil be used later? However, it's a bit more clear to have two separate parameters, instead of a tuple, since the tuple is not optional.